org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeBuilder.java

```diff
@@ -0,0 +1,71 @@
+package org.code.javabuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+
+/** The orchestrator for code compilation and execution. */
+public class CodeBuilder {
+  private final OutputAdapter outputAdapter;
+  private final InputAdapter inputAdapter;
+  private final ProjectFileManager fileManager;
+  private final File tempFolder;
+
+  public CodeBuilder(
+      InputAdapter inputAdapter, OutputAdapter outputAdapter, ProjectFileManager fileManager)
+      throws UserFacingException {
+    this.outputAdapter = outputAdapter;
+    this.inputAdapter = inputAdapter;
+    this.fileManager = fileManager;
+    try {
+      tempFolder = Files.createTempDirectory("tmpdir").toFile();
+    } catch (IOException e) {
+      throw new UserFacingException(
+          "We hit an error on our side while loading your program. Try again.");
+    }
+  }
+
+  /**
+   * Loads and compiles the user's code
+   *
+   * @throws UserFacingException if the user's code contains a compiler error or if we are unable to
+   *     compile due to internal errors.
+   */
+  public void compileUserCode() throws UserFacingException {
```

Comment:
This is fine as-is now, but I think at some point it would be better for this method to return something that represents the compiled code and for runUserCode to take that as input.  (I still expect compile and run to be managed by two separate classes at some point.  These are two fairly different operations that have a very clear interface.)

Reply:
Good point. I'll leave it as is for now, but we'll need to keep that in mind - especially if we split compiling and running into different lambdas.

Reply:
something else to consider on this side is we do have a use case for just compiling code and giving a thumbs up or compilation error message. This will happen with commit code/save to backpack. Probably don't need to change it now but I think two separate classes will eventually make sense.

Reply:
👍